### PR TITLE
JUCX: create endpoint functionality.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
UCP endpoint in JUCX.

## Why ?
To be able to communicate in JUCX.

## How ?
Callback functionality (accept, connect handlers, etc.) would be next steps.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9742/ for details (Mellanox internal link).

File: bindings/java/src/main/native/endpoint.cc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        ep_params.flags = env->GetLongField(ucp_ep_params, field);
+    }
+
+    if (ep_params.field_mask & UCP_EP_PARAM_FIELD_SOCK_ADDR) {
+        struct sockaddr_storage worker_addr;
+        socklen_t addrlen;
+        memset(&worker_addr, 0, sizeof(struct sockaddr_storage));
+
+        field = env->GetFieldID(ucp_ep_params_class,
+                            "socketAddress", "Ljava/net/InetSocketAddress;");
```
Comment (yosefe) [inline bindings/java/src/main/native/endpoint.cc]: indentation

File: bindings/java/src/test/java/org/ucx/jucx/UcpEndpointTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        while (!success && interfaces.hasMoreElements())
+        {
+            NetworkInterface networkInterface = interfaces.nextElement();
+            Enumeration<InetAddress> inetAddresses = networkInterface.getInetAddresses();
+            while (inetAddresses.hasMoreElements()) {
+                InetAddress inetAddress = inetAddresses.nextElement();
+                if (!inetAddress.isLoopbackAddress()) {
+                    try {
+                        InetSocketAddress addr = new InetSocketAddress(inetAddress,
+                          UcpListenerTest.port);
```
Comment (yosefe) [inline bindings/java/src/test/java/org/ucx/jucx/UcpEndpointTest.java]: indent on column (what is the current style for java)?
also in other places in this function

Comment (petro-rudenko) [reply bindings/java/src/test/java/org/ucx/jucx/UcpEndpointTest.java]: For java we're using 4 spaces indents: https://github.com/openucx/ucx/blob/master/bindings/java/checkstyle.xml#L61

**Comment by petro-rudenko** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9779/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:mlx:retest

**Comment by yosefe** [comment]
@petro-rudenko pls use "JUCX: " prefix for commit title (would need force-push to fix it)

**Comment by petro-rudenko** [comment]
@yosefe done.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9799/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9803/ for details (Mellanox internal link).

**Comment by petro-rudenko** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9809/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9811/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9826/ for details (Mellanox internal link).

**Pull request merged**